File: backEnd/src/main/java/com/quiz/ourclass/domain/board/repository/CommentRepository.java
package com.quiz.ourclass.domain.board.repository;

import com.quiz.ourclass.domain.board.entity.Comment;
import com.quiz.ourclass.domain.board.entity.Post;
import com.quiz.ourclass.domain.member.entity.Member;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;

public interface CommentRepository extends JpaRepository<Comment, Long> {

    List<Comment> findAllByPostId(Long postId);

    void deleteByParentId(Long parentId);

    void deleteByPostId(Long postId);

    boolean existsByPostAndMember(Post post, Member member);

    boolean existsByParentIdAndMember(Long parentId, Member member);
}

File: android/domain/src/main/java/com/sixkids/domain/repository/CommentRepository.kt
package com.sixkids.domain.repository

interface CommentRepository {
    suspend fun createComment(postId: Long, content: String, parentId: Long): Long

    suspend fun deleteComment(id: Long): Boolean

    suspend fun updateComment(id: Long, content: String): Long

    suspend fun reportComment(id: Long): Boolean
}